src/AppInstallerCLITests/RestInterface_1_0.cpp

```diff
@@ -0,0 +1,413 @@
+            REQUIRE(actualInstaller.Switches.at(InstallerSwitchType::InstallLocation) == "C:\\Users\\User1");
+            REQUIRE(actualInstaller.Switches.at(InstallerSwitchType::Log) == "/l");
+            REQUIRE(actualInstaller.Switches.at(InstallerSwitchType::Update) == "/u");
+            REQUIRE(actualInstaller.Switches.at(InstallerSwitchType::Custom) == "/custom");
+            REQUIRE(actualInstaller.InstallerSuccessCodes.size() == 1);
+            REQUIRE(actualInstaller.InstallerSuccessCodes.at(0) == 0);
+            REQUIRE(actualInstaller.UpdateBehavior == UpdateBehaviorEnum::Install);
+            REQUIRE(actualInstaller.Commands.at(0) == "command1");
+            REQUIRE(actualInstaller.Protocols.at(0) == "protocol1");
+            REQUIRE(actualInstaller.FileExtensions.at(0) == ".fileextension");
+            REQUIRE(actualInstaller.Dependencies.WindowsFeatures.at(0) == "feature1");
+            REQUIRE(actualInstaller.Dependencies.WindowsLibraries.at(0) == "library1");
+            REQUIRE(actualInstaller.Dependencies.PackageDependencies.at(0).Id == "Foo.Baz");
+            REQUIRE(actualInstaller.Dependencies.PackageDependencies.at(0).MinVersion == "2.0.0");
+            REQUIRE(actualInstaller.Dependencies.ExternalDependencies.at(0) == "FooBarBaz");
+            REQUIRE(actualInstaller.PackageFamilyName == "FooBar.PackageFamilyName");
+            REQUIRE(actualInstaller.ProductCode == "");
+            REQUIRE(actualInstaller.Capabilities.at(0) == "bluetooth");
+            REQUIRE(actualInstaller.RestrictedCapabilities.at(0) == "restrictedCapability");
+        }
+    };
+}
+
+TEST_CASE("Search_GoodResponse", "[RestSource]")
+{
+    utility::string_t sample = _XPLATSTR(
+        R"delimeter({
+            "Data" : [
+               {
+              "PackageIdentifier": "git.package",
+              "PackageName": "package",
+              "Publisher": "git",
+              "Versions": [
+                {   "PackageVersion": "1.0.0" },
+                {   "PackageVersion": "2.0.0"}]
+            }]
+        })delimeter");
+
+    HttpClientHelper helper{ GetTestHandler(web::http::status_codes::OK, std::move(sample)) };
+    Interface v1{ TestRestUriString, std::move(helper) };
+    Schema::IRestClient::SearchResult searchResponse = v1.Search({});
+    REQUIRE(searchResponse.Matches.size() == 1);
+    Schema::IRestClient::Package package = searchResponse.Matches.at(0);
+    REQUIRE(package.PackageInformation.PackageIdentifier.compare("git.package") == 0);
+    REQUIRE(package.PackageInformation.Publisher.compare("git") == 0);
+    REQUIRE(package.PackageInformation.PackageName.compare("package") == 0);
+    REQUIRE(package.Versions.size() == 2);
+    REQUIRE(package.Versions.at(0).VersionAndChannel.GetVersion().ToString().compare("1.0.0") == 0);
+    REQUIRE(package.Versions.at(1).VersionAndChannel.GetVersion().ToString().compare("2.0.0") == 0);
+}
+
+TEST_CASE("Search_BadResponse", "[RestSource]")

```

Comment:
>Search_BadResponse [](start = 11, length = 18)

nit: if we don't want to write comments here, we should make the test name with a little more info for better readability
Search_BadResponse_<WhyBad>